cpp/include/kvikio/file_handle.hpp

```diff
@@ -496,6 +496,79 @@ class FileHandle {
     return parallel_io(op, devPtr_base, size, file_offset, task_size, devPtr_offset);
   }
 
+  /**
+   * @brief Reads specified bytes from the file into the device memory.
+   *
+   * This API reads asynchronously the data from the GPU memory to the file at a specified offset
+   * and size bytes by using GDS functionality. The API works correctly for unaligned
+   * offset and data sizes, although the performance is not on-par with aligned read.
+   * This is an asynchronous call and will be executed in sequence for the specified stream.
+   *
+   * @note For the `devPtr_offset`, if data will be read starting exactly from the
+   * `devPtr_base` that is registered with `buffer_register`, `devPtr_offset` should
+   * be set to 0. To read starting from an offset in the registered buffer range,
+   * the relative offset should be specified in the `devPtr_offset`, and the
+   * `devPtr_base` must remain set to the base address that was used in the
+   * `buffer_register` call.
+   *
+   * @param devPtr_base Base address of buffer in device memory. For registered buffers,
+   * `devPtr_base` must remain set to the base address used in the `buffer_register` call.
+   * @param size Size in bytes to read.
+   * @param file_offset Offset in the file to read from.
+   * @param devPtr_offset Offset relative to the `devPtr_base` pointer to read into.
+   * This parameter should be used only with registered buffers.
```

Comment:
Is there a way to distinguish registered buffers in the type system at the moment? That would be preferable over this `void *` C-like interface.

Reply:
No, not ATM but it would be a cool addition: https://github.com/rapidsai/kvikio/issues/266